### PR TITLE
Cohort services

### LINKED ISSUE #350: Generate the services to interact with the cohorts
### Feature description

Generate the services to be able to interact with the `cohorts` entity:

- [x] Create a cohort
- [x] Update a cohort
- [x] Get those cohorts related with a course
- [x] Setup the correct `cohort` permissions in the API

### PR DESCRIPTION
**Pull request by Tansito**
## Changes

Fix #350

This PR adds the service to interact with the `cohort` from `saiba`. To be more specific:
- create a cohort
- update a cohort
- get cohorts from a specific course

## Implementation details

In this PR we are generating the next things:

- The service for the `cohort` end-points.
- In the service how we need to use the API Rest due to the M2M relationship to write things we added some configurations similar to GraphQL: **[more info here](https://github.com/directus/directus/discussions/11773)**
- Added different transformation methods to manage the transition from **DTO** to saiba's **type**.

### CODE REVIEW

File: services/cohort.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import {
+  getGraphQLURL,
+  gqlQueryOptions,
+  getItemsURL,
+  apiRestOptions,
+} from './utils';
+import { Query } from '@/store/types';
```
Comment (y4izus) [inline services/cohort.ts:7]:
```suggestion
import type { Query } from '@/store/types';
```

File: services/cohort.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    getGraphQLURL(),
+    gqlQueryOptions(query, queryVariables)
+  );
+
+  return res.data;
+};
+
+export const createCohort = async (
+  courseId: Course['id'],
+  name: Cohort['name'],
```
Comment (y4izus) [inline services/cohort.ts:46]: What about passing directly the Cohort object?

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you very much! some small changes and that's all :D

File: services/cohort.ts (hunk 2 above)
Comment (Tansito) [reply services/cohort.ts:46]: @korgan00 and me commented the same thing but we thought that would be better to create an issue and do the refactor in all the services (we are using the same approach in all the services), I don't know what you think.

File: services/cohort.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const res = await $fetch<CohortDTO>(
+    COHORTS_URL,
+    apiRestOptions('POST', body)
+  );
+
+  return res;
+};
+
+export const updateCohort = async (
+  cohortId: Cohort['id'],
```
[the remark this answers is not in the document]
Comment (Tansito) [reply services/cohort.ts:86]: Same as my previous comment.

File: services/cohort.ts (hunk 2 above)
Comment (y4izus) [reply services/cohort.ts:46]: you're right! you already told me about that but I forget it, sorry!🤦🏽‍♀️.  I'm completely agree :D

File: store/cohort/utils.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    id,
+    name,
+    invitations,
+    members: members.map((member) => memberDTOToMembers(member)),
+    autoEnrollment: auto_enrollment,
+    enrollmentCode: enrollment_code,
+  };
+}
+
+function memberDTOToMembers(memberDTO: MemberDTO): Member {
```
Comment (korgan00) [inline store/cohort/utils.ts:18]:
```suggestion
function memberDTOToMember(memberDTO: MemberDTO): Member {
```

**Review by korgan00 — APPROVED** [review]
LGTM, minor change

File: services/cohort.ts (hunk 1 above)
Comment (Tansito) [reply services/cohort.ts]: Done in [308f370](https://github.com/Qiskit/saiba/pull/349/commits/308f370661b5098a5670c2663fe85202103923f8)

File: services/utils.ts [hunk 5]
```diff
@@ -6,10 +6,17 @@ export const getGraphQLURL = () => {
   return `${runtimeConfig.apiBase}/graphql`;
 };
 
+export const getItemsURL = () => {
+  const runtimeConfig = useRuntimeConfig();
+
+  return `${runtimeConfig.apiBase}/items`;
+};
```
[the remark this answers is not in the document]
Comment (Tansito) [reply services/utils.ts]: Done in [8d06fc1](https://github.com/Qiskit/saiba/pull/349/commits/8d06fc19f0e9ad34713107a7b0d0dd64f802798c)

File: store/cohort/utils.ts (hunk 4 above)
Comment (Tansito) [reply store/cohort/utils.ts]: Done in [8c48ef1](https://github.com/Qiskit/saiba/pull/349/commits/8c48ef1981c827fcbeb147110f4ce2481d471756)

**Review by y4izus — APPROVED** [review]

**Pull request merged**